wazo_confd/plugins/meeting/resource.py

```diff
@@ -0,0 +1,258 @@
+        user_service,
+        tenant_service,
+        endpoint_sip_service,
+        endpoint_sip_template_service,
+        hostname,
+        port,
+    ):
+        super().__init__(service)
+        self._user_service = user_service
+        self._tenant_service = tenant_service
+        self._endpoint_sip_service = endpoint_sip_service
+        self._endpoint_sip_template_service = endpoint_sip_template_service
+        self._init_schema(hostname, port)
+
+    def build_headers(self, meeting):
+        return {'Location': url_for('meetings', uuid=meeting.uuid, _external=True)}
+
+    @required_acl('confd.meetings.create')
+    def post(self):
+        return self._post(request.get_json())
+
+    def _post(self, body):
+        form = self.schema().load(body)
+        form['uuid'] = uuid4()
+        form = self.add_tenant_to_form(form)
+        form = self.find_owners(form)
+        form = self.add_endpoint_to_form(form)
+        model = self.model(**form)
+        model = self.service.create(model)
+        return self.schema().dump(model), 201, self.build_headers(model)
+
+    @required_acl('confd.meetings.read')
+    def get(self):
+        return super().get()
+
+    def add_endpoint_to_form(self, form):
+        tenant = self._tenant_service.get(form['tenant_uuid'])
+        template_uuid = tenant.meeting_guest_sip_template_uuid
+        if not template_uuid:
+            logger.warning('Cannot create guest endpoint, missing template')
+            return form
+
+        template = self._endpoint_sip_template_service.get(template_uuid)
+        endpoint_name = endpoint_username = context = 'wazo-meeting-{}-guest'.format(
+            form['uuid']
+        )
+        endpoint_body = {
+            'name': endpoint_name,
+            'tenant_uuid': form['tenant_uuid'],
+            'label': 'External meeting guest {}'.format(form['name']),
+            'templates': [template],
+            'auth_section_options': [
+                ['username', endpoint_username],
+                ['password', random_string(16)],
+            ],
+            'aor_section_options': [],
+            'endpoint_section_options': [
+                ['context', context],
+            ],
+            'identify_section_options': [],
+            'registration_section_options': [],
+            'registration_outbound_auth_section_options': [],
+            'outbound_auth_section_options': [],
+        }
+        endpoint_model = EndpointSIP(**endpoint_body)
+        endpoint = self._endpoint_sip_service.create(endpoint_model)
+        form['guest_endpoint_sip_uuid'] = endpoint.uuid
+        return form
+
+    def find_owners(self, form):
+        owner_uuids = form.pop('owner_uuids', None) or []
+        owners = []
+        tenant_uuid = form['tenant_uuid']
+        for uuid in owner_uuids:
+            owners.append(self._user_service.get(uuid, tenant_uuids=[tenant_uuid]))
+        form['owners'] = owners
+        return form
+
+    def _find_user_uuid(self):
```

Comment:
This is going to need some factorization: wazo-calld has the same code for the same purpose.